### ISSUE TITLE
operator ireduce duplicates merge '<<:' operation when used with multiplication 'ireduce({}; . * $item)'

### ISSUE THREAD

**Issue by kuvaldini** [mikefarah/yq#899]
**Describe the bug**
$subj

Version of yq: 4.11.1
Operating system: mac
Installed via: homebrew

**Input Yaml**
data1.yml:
```yaml
jobs:
  build-iroha-ubuntu-debug:
    steps:
      - &step_export_cxx
        env: &step_export_cxx_env
          CC: ${{matrix.CC}}
          CCACHE_PATH: /usr/lib/ccache

  build-iroha-macos:
    steps:
      - <<: *step_export_cxx
        env:
          <<: *step_export_cxx_env
          CCACHE_PATH: /usr/local/opt/ccache/libexec
```

**Command**
The command you ran:
```
yq eval '. as $item ireduce ({}; . * $item)' data1.yml
```

**Actual behavior**

```yaml
jobs:
  build-iroha-ubuntu-debug:
    steps:
      - &step_export_cxx
        env: &step_export_cxx_env
          CC: ${{matrix.CC}}
          CCACHE_PATH: /usr/lib/ccache
  build-iroha-macos:
    steps:
      - !!merge <<: *step_export_cxx
        env:
          !!merge <<: *step_export_cxx_env
          CCACHE_PATH: /usr/local/opt/ccache/libexec
          <<: *step_export_cxx_env
        <<: *step_export_cxx
```

**Expected behavior**

```yaml
jobs:
  build-iroha-ubuntu-debug:
    steps:
      - &step_export_cxx
        env: &step_export_cxx_env
          CC: ${{matrix.CC}}
          CCACHE_PATH: /usr/lib/ccache
  build-iroha-macos:
    steps:
      - <<: *step_export_cxx
        env:
          <<: *step_export_cxx_env
          CCACHE_PATH: /usr/local/opt/ccache/libexec
```

**Comment by kuvaldini** [comment]
`yq eval '. as $item ireduce ({}; . *+ $item)' data1.yml` solves, note `*+`

but there was a problem when it

**Comment by mikefarah** [comment]
Yep that's a bug - will investigate

**Comment by kuvaldini** [comment]
## WORKAROUND. Use YQ version 3
Install on Mac

    wget https://github.com/mikefarah/yq/releases/download/3.4.1/yq_darwin_amd64 -O /usr/local/bin/yq-3 &&
      chmod +x /usr/local/bin/yq-3

Install on Linux

    wget https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64 -O /usr/local/bin/yq-3 &&
      chmod +x /usr/local/bin/yq-3

Run

    yq-3 merge *.yaml |
      yq-3 read --explodeAnchors -

yq version 3 works much more faster, at least 50 times. even with pipe

**Comment by mikefarah** [comment]
I've fixed and released the bug in 4.11.2, I haven't been able to replicate that performance issue though - in my testing with large files v4 is faster - I don't support you're able to give me a sample file to replicate that issue? The bug may have been slowing it down - I'd be interested to see if you still get the same results.

**Comment by kuvaldini** [comment]
Thank you. I will check it next time coming to yamls.
